linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml

```diff
@@ -426,4 +432,17 @@
     - "{{ res_count }}"
   loop_control:
     loop_var: data
-  when: not async and uri_hostname != 'localhost'
+  when:
+    - not async and uri_hostname != 'localhost'
+    - extract_ip_address_result_remote is defined and extract_ip_address_result.changed == true
+
+- name: "Append to topology_outputs_libvirt_nodes"
+  set_fact:
+    topology_outputs_libvirt_nodes: "{{ topology_outputs_libvirt_nodes + [ { 'name': data[0], 'resource_type': 'libvirt_res' }] }}"
+  with_nested:
+    - "{{ output_name }}"
+  loop_control:
+    loop_var: data
+  when:
+    - extract_ip_address_result is not defined
+    - extract_ip_address_result is not defined
```

Comment:
@abraverm isn't this a duplicate condition in when statement?

Reply:
you are right, it should be extract_ip_address_result_remove